scripts/test_matrix.py

```diff
@@ -0,0 +1,190 @@
+	['scalable_ep',			{'ofi-rxm': 0, 'psm': 0, 'psm2': 0, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	['cmatose',			prov_msg],
+	['rdm_shared_av',		{'ofi-rxm': 0, 'psm': 0, 'psm2': 0, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+]
+
+standard = [
+	['msg_pingpong',		prov_msg],
+	['msg_pingpong -v',		prov_msg],
+	['msg_pingpong -k',		prov_msg],
+	['msg_pingpong -k -v',		prov_msg],
+	['msg_bw',			prov_msg],
+	['msg_bw -v',			prov_msg],
+	['msg_bw -k',			prov_msg],
+	['msg_bw -k -v',		prov_msg],
+	['rdm_pingpong',		prov_rdm],
+	['rdm_pingpong -v',		prov_rdm],
+	['rdm_pingpong -k',		prov_rdm],
+	['rdm_pingpong -k -v',		prov_rdm],
+	['rdm_tagged_pingpong',		prov_rdm],
+	['rdm_tagged_pingpong -v',	prov_rdm],
+	['rdm_tagged_pingpong -k',	prov_rdm],
+	['rdm_tagged_pingpong -k -v',	prov_rdm],
+	['rdm_tagged_bw',		prov_rdm],
+	['rdm_tagged_bw -v',		prov_rdm],
+	['rdm_tagged_bw -k',		prov_rdm],
+	['rdm_tagged_bw -k -v',		prov_rdm],
+	# psm, psm2 support FI_EP_DGRAM but the dgram_pingong test doesn't support FI_CONTEXT
+	['dgram_pingpong',		{'ofi-rxm': 0, 'psm': 0, 'psm2': 0, 'sockets': 1, 'udp': 1, 'verbs': 0}],
+	['dgram_pingpong -v',		{'ofi-rxm': 0, 'psm': 0, 'psm2': 0, 'sockets': 1, 'udp': 1, 'verbs': 0}],
+	['dgram_pingpong -k',		{'ofi-rxm': 0, 'psm': 0, 'psm2': 0, 'sockets': 1, 'udp': 1, 'verbs': 0}],
+	['dgram_pingpong -k -v',	{'ofi-rxm': 0, 'psm': 0, 'psm2': 0, 'sockets': 1, 'udp': 1, 'verbs': 0}],
+	['rma_bw -e msg -o write',	prov_msg],
+	['rma_bw -e msg -o read',	prov_msg],
+	['rma_bw -e msg -o writedata',	prov_msg],
+	# rma_bw test doesn't support FI_CONTEXT required by psm, psm2
+	['rma_bw -e rdm -o write',	{'ofi-rxm': 1, 'psm': 0, 'psm2': 0, 'sockets': 1, 'udp': 0, 'verbs': 1}],
+	['rma_bw -e rdm -o read',	{'ofi-rxm': 1, 'psm': 0, 'psm2': 0, 'sockets': 1, 'udp': 0, 'verbs': 1}],
+	# verbs/RDM doesn't support fi_writedata
+	['rma_bw -e rdm -o writedata',	{'ofi-rxm': 1, 'psm': 0, 'psm2': 0, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	['msg_rma -o write',		prov_msg],
+	['msg_rma -o read',		prov_msg],
+	['msg_rma -o writedata',	prov_msg],
+	['msg_stream',			prov_msg],
+	['rdm_atomic -o all -I 1000',	{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	# rdm_cntr_pingong doesn't support FI_CONTEXT which is required by psm, psm2
+	['rdm_cntr_pingpong',		{'ofi-rxm': 0, 'psm': 0, 'psm2': 0, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	['rdm_multi_recv',		{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 1}],
+	['rc_pingpong',			prov_msg],
+	['rc_pingpong -e',		prov_msg],
+]
+
+short = []
+for test in standard:
+	short.append([test[0] + ' -I 5', test[1]])
+
+complex = [
+	['ubertest', {'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 1, 'verbs': 1}],
+]
+
+tests = {
+	'unit'		: unit,
+	'neg_unit'	: neg_unit,
+	'simple'	: simple,
+	'short'		: short,
+	'standard'	: standard,
+	'complex'	: complex,
+	}
+
+if __name__ == '__main__':
+	if len(sys.argv) != 3:
```

Comment:
You should use argparse and set a choices option for the provider and test type. 

```python
import argparse

parser = argparse.ArgumentParser(description='Returns a comma separated list of tests for requested test type')
parser.add_argument('provider', choices=['all', 'ofi-rxm', 'psm', 'psm2', 'sockets', 'verbs', 'udp'],
                    help='Generate tests for given provider, all returns all tests independent of the provider')
parser.add_argument('test_type', choices=['unit', 'neg_unit', 'simple', 'short', 'standard', 'complex']
                    help='Generate tests using given test_type')
args = parser.parse_args()
```